cmd/kubeadm/app/phases/kubelet/config.go

```diff
@@ -66,7 +68,20 @@ func WriteConfigToDisk(cfg *kubeadmapi.ClusterConfiguration, kubeletDir, patches
 		}
 	}
 
-	return writeConfigBytesToDisk(kubeletBytes, kubeletDir)
+	if features.Enabled(cfg.FeatureGates, features.NodeLocalCRISocket) {
+		kubeletBytes, err = applyKubeletConfigPatchFromFile(kubeletBytes, filepath.Join(kubeletDir, kubeadmconstants.KubeletInstanceConfigurationFileName), output)
+		if err != nil {
+			return errors.Wrap(err, "could not apply patches to the KubeletConfiguration")
+		}
```

Comment:
```suggestion
		file := filepath.Join(kubeletDir, kubeadmconstants.KubeletInstanceConfigurationFileName)
		kubeletBytes, err = applyKubeletConfigPatchFromFile(kubeletBytes, file, output)
		if err != nil {
			return errors.Wrapf(err, "could not apply kubelet instance configuration as a patch from %q", file)
		}
```

to make the error different from the one on line 65